src/main/java/org/broadinstitute/hellbender/utils/SequenceDictionaryUtils.java

```diff
@@ -308,21 +318,32 @@ private static boolean commonContigsHaveSameLengths(Set<String> commonContigs, S
 
     /**
      * Helper routine that returns whether two sequence records are equivalent, defined as having the same name and
-     * lengths
+     * lengths.
+     *
+     * NOTE: we allow the lengths to differ if one or both are UNKNOWN_SEQUENCE_LENGTH
      *
      * @param first first sequence record to compare
      * @param second second sequence record to compare
      * @return true if first and second have the same names and lengths, otherwise false
      */
-    private static boolean sequenceRecordsAreEquivalent(final SAMSequenceRecord first, final SAMSequenceRecord second) {
+    @VisibleForTesting
+    static boolean sequenceRecordsAreEquivalent(final SAMSequenceRecord first, final SAMSequenceRecord second) {
         if ( first == second ) {
             return true;
         }
         if ( first == null || second == null ) {
             return false;
         }
+        if (! first.getSequenceName().equals(second.getSequenceName())){
+            return false;
+        }
+        final int length1 = first.getSequenceLength();
+        final int length2 = second.getSequenceLength();
 
-        return first.getSequenceLength() == second.getSequenceLength() && first.getSequenceName().equals(second.getSequenceName());
+        if (length1 != length2 && length1 != SAMSequenceRecord.UNKNOWN_SEQUENCE_LENGTH && length2 != SAMSequenceRecord.UNKNOWN_SEQUENCE_LENGTH){
+            return false;
+        }
+        return true;
```

Comment:
I'm not sure we should be returning true if one of the lengths is unknown. Can you explain?